src/GitHub/Data/Request.hs

```diff
@@ -122,6 +123,7 @@ data Request (k :: Bool) a where
     PagedQuery   :: FromJSON (Vector a) => Paths -> QueryString -> Maybe Count -> Request k (Vector a)
     Command      :: FromJSON a => CommandMethod a -> Paths -> LBS.ByteString -> Request 'True a
     StatusQuery  :: StatusMap a -> Request k () -> Request k a
+    HeaderQuery  :: FromJSON a => Types.RequestHeaders -> Request k a -> Request k a
```

Comment:
Clever!

is `FromJSON a` required here, it shouldn't be?

note to myself: refactor the `Request` so we cannot have multiple stacked `HeaderQuery` or `StatusQuery` as it doesn't make sense (or make sure that the code recurse properly).